t/24-worker-jobs.t

```diff
@@ -1081,6 +1081,26 @@ subtest 'Dynamic schedule' => sub {
     $job->_upload_results_step_0_prepare(sub { });
     is_deeply $job->{_test_order}, $test_order, 'Initial test schedule';
 
+    # do not read test_order.json when current_test is null
+    $autoinst_status->{current_test} = '';
+    $status_file->spurt(encode_json($autoinst_status));
+    $job->_upload_results_step_0_prepare(sub { });
+
+    $job_mock->redefine(
+        _upload_results_step_1_post_status => sub {
+            my ($self, $status, $callback) = @_;
+            my $job_id = $self->id;
+            $self->client->send(
+                post         => "jobs/$job_id/status",
+                json         => {status => $status},
+                non_critical => $self->is_stopped_or_stopping,
+                callback     => $callback,
+            );
+            return $status;
+        });
```

Comment:
It looks like this does exactly what the actual function does. So better just use `unmock('_upload_results_step_1_post_status')`.

Reply:
I added one line `return $status`.  The `$status` will include the `test_order` when reloading test_order.json, so I use it to check if the reloading operation is done.

Reply:
The real function doesn't return anything, though? :thinking:

How about a `combined_unlike { ... } qr/Test schedule has changed, reloading test_order.json/` and a `combined_like { ... } qr/Test schedule has changed, reloading test_order.json/` in the case after this one (where we expected a change)?

Reply:
Duplicating the original code is a bad idea. Use a coderef to the original function if you have to test it this way.

Reply:
> The real function doesn't return anything, though?

The real function returns what `$self->client->send` returns but that is likely never really used. So it would be possibly to simply adjust the real function.

Alternatively, you could also use the following code (not tested):

```
    $job_mock->redefine(
        _upload_results_step_1_post_status => sub {
            my ($self, $status, $callback) = @_;
            $job_mock->original('_upload_results_step_1_post_status')->($self, $status, $callback);
            return $status;
        });
```

Reply:
@kalikiana I changed the code to use `combined_unlike` and `combined_like`. Thanks.
@kraih Could you give me an example about how to use a coderef ? Thanks.
@Martchus The way you gave worked well, I tested it. And I decided to use `combined_unlike` because it do not need to redefine other function. Thanks.

Reply:
You can always get a coderef for a function with `my $coderef = $class->can('foo')` and then call it with `$coderef->(...)`. But the way @Martchus showed is better here anyway.